extensions/src/ACRE2Arma/arma2ts/CallExt_DllMain.cpp

```diff
@@ -188,33 +187,43 @@ void __stdcall RVExtension(char *output, int outputSize, const char *function)
         case PIPE_COMMAND_READ: {
             if (readConnected) {
                 DWORD cbRead;
-                DWORD err;
                 BOOL ret;
-                char *value = new char[4096];
+                constexpr size_t read_length = 4097U;
+                char value[read_length]; // Allocate on stack to delegate memory management to compiler,
+                // allows up to 4096 char string (+1 to ensure NUL terminated), like initial version
+
                 //DEBUG("Read from pipe [%d]\n", hPipe);
-                ret = ReadFile(readHandle, (LPVOID)value, 4096, &cbRead, NULL);
-                if ( ! ret) {
-                    err = GetLastError();
-                    if (err == ERROR_BROKEN_PIPE) {
-                        ClosePipe();
-                    }
+                ret = ReadFile(readHandle, (LPVOID)value, sizeof(char) * (read_length - 1U), &cbRead, NULL);
+                if (!ret) {
+                    DWORD err = GetLastError();
                     //DEBUG("ReadFile failed, [%08x]\r\n", err);
-                }
-                if (cbRead != 0) {
-                    //DEBUG("Read data: %s\n", value);
-                    strncpy(output,value,cbRead);
-                } else {
-                    if (err == 232) {
-                        strncpy(output,"_JERR_NULL",outputSize);
+                    if (err == ERROR_NO_DATA) {
+                        strncpy(output, "_JERR_NULL", outputSize);
                     } else {
-                        //LOG("PIPE ERROR: %d\r\n", err);
-                        strncpy(output,"_JERR_FALSE",outputSize);
+                        if (err == ERROR_BROKEN_PIPE) {
+                            ClosePipe();
+                        }
+                        strncpy(output, "_JERR_FALSE", outputSize);
                     }
+                } else if (cbRead != 0) {
+                    //DEBUG("Read data: %s\n", value);
+
+                    if (cbRead >= (size_t)outputSize) {
+                        // Prevent buffer overflow
+                        cbRead = outputSize - 1U;
+                    }
+
+                    // Ensure NUL terminated string (required by strncpy_s)
```

Comment:
This comment is no longer needed. Do you still need to  add then the '\0'  at the end of the string?

Reply:
`strncpy` also required NUL terminated string, comment updated